Enhancement/#9 implementation application layer

Yared-H opened the pull request.

Commit:
Delete aplicationHeader.c

Commit:
Add files via upload

Commit:
Delete applicationHeader.h

Commit:
Add files via upload

Commit:
Add files via upload

cvonder1 commented on src/applicationLayer.c:
```diff
@@ -0,0 +1,42 @@
+
+Result put(char* key, char* value ){
+    Result result;
+    if(save(key,value)==0){
+        result.value = value;
+        result.error_code = save(key, value);
+        return result;
+    }
+    else{
+        result.error_code = save(key, value);
+        return result;
+    }
```
One call to save is sufficient. Moreover multiple calls introduces ambiguity, because the state might have changed in between.

With the new Result structure, the value string has a fixed length. So `strncpy` would be better than the assignment. But I guess this is alright for now.

cvonder1 commented on include/datenhaltung.h:
```diff
@@ -0,0 +1,20 @@
+#ifndef DATEN_HALTUNG_HEADER
+#define DATEN_HALTUNG_HEADER
+// public
+/*
+    In Result.value steht das value des keys, wenn er gefunden wurde
+    ansonsten bleibt value leer und in Result.error_code steht KEY_NOT_FOUND
+*/
+Result find_by_key(char* key);
+
+/*
+    Gibt SUCCESS oder eine Fehlermeldung über 1000 zurück
+*/
+int save(char* key, char* value);
+
+/*
+    Gibt SUCCESS oder eine Fehlermeldung über 1000 zurück
+*/
+int delete_by_key(char* key);
+
+#endif
```
When I see this correctly, `datenhaltung.h` is just copy&pasted from the main branch. This causes a merge conflict and is generally not a good idea.
It is better to merge or cherry-pick the desired changes.
So you need to revert commit 75d7f42e and merge the main branch into this one.

cvonder1 commented on include/applicationLayer.h:
```diff
@@ -0,0 +1,12 @@
+       bleibt LEER.
+*/
+//Entscheidet ob und welcher Befehl ausgef�hrt wird, gibt Result an ClientSession zur�ck. M�gliche Error: COMMAND_NOT_FOUND
+Result executeCommand(Command command);
```
Sry that I didn't pay attention to this before. But the applicationLayer header depends on the utils header, because it uses both structs `Result` and `Command`. So for the sake of completeness this header should include the utils header.

cvonder1 reviewed (changes requested):
All edited files use a CRLF line ending, please change the line ending to LF.

Btw. the commit messages say absolutely nothing about the commits.

Commit:
took out datenhaltung.h

Ya-mi-ha committed:
Changed line endings to LF

Commit:
Merge branch 'main' into enhancement/#9-implementation_application_layer

Commit:
Fix compiler issues

Commit:
Merge branch 'dev' into enhancement/#9-implementation_application_layer

cvonder1 reviewed (commented):
I don't know what those two files `misc.xml` and `vcs.xml` do, but I have the feeling that they are specific for your local machine. As far as I know @lmaraite uses CLion. Maybe you know something about it?

In either case it is better to open a new issue/pull request for the changes, because they are not related to the application layer implementation.

I will quickly remove them, so that I can merge this branch into dev. You can open a new pull request for further discussion.

Commit:
fix: remove .idea

idea specific configurations have nothing to do with the application layer
implementation. Please reapply those changes on an appropiate branch.
